### PR TITLE
Update stitch

### PR DESCRIPTION
**Pull request by LouisLeNezet**
Add nf-test for stitch modules

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/stitch/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,178 @@
+def pathbam = "file('https://github.com/nf-core/test-datasets/raw/modules/data/genomics/homo_sapiens/illumina/"
```
Comment (maxulysse) [inline modules/nf-core/stitch/tests/main.nf.test:1]: can you use the params modules_test data path thingy?

File: modules/nf-core/stitch/main.nf [hunk 2]
```diff
@@ -8,8 +8,8 @@ process STITCH {
         'biocontainers/r-stitch:1.6.10--r43h06b5641_0' }"
 
     input:
-    tuple val(meta) , path(posfile), path(input, stageAs: "input"), path(rdata, stageAs: "RData_in"), val(chromosome_name), val(K), val(nGen)
-    tuple val(meta2), path(collected_crams), path(collected_crais), path(cramlist)
+    tuple val(meta), path(collected_crams), path(collected_crais), path(cramlist)
```
Comment (matthdsm) [inline modules/nf-core/stitch/main.nf:11]: Why the swap?

Comment (LouisLeNezet) [reply modules/nf-core/stitch/main.nf:11]: It was to be consistent with other imputation modules.
The input / target files first then the reference panel.

Comment (matthdsm) [reply modules/nf-core/stitch/main.nf:11]: Agreed then 👍🏻

**Review by matthdsm — APPROVED** [review]

**Pull request merged**